### PR TITLE
0.1.0dev

### LINKED ISSUE #20: more descriptive error for failing capitains guidelines
Currently says "Text can't be read through Capitains standards"

Would be good if it was more explicit about *which* standards failed (e.g. wrong filename, etc.)

### LINKED ISSUE #23: validate urn in textgroup level __cts__.xml
Hook passed my textgroup __cts__.xml even though the textgroup element had a urn that didn't correspond to the directory structure.  Is there any testing of urns vs file structure ?

### LINKED ISSUE #24: Check for edition/translation lang attribute

### LINKED ISSUE #31: Shows wrong number of passages (2 instead of 21)
Example data [data.zip](https://github.com/Capitains/HookTest/files/269175/data.zip)

### LINKED ISSUE #35: Test on TEI Body for URN is wrong.
Guidelines and test are not the same. See what would need to change.

### LINKED ISSUE #36: Move to MyCapytain > 1.0.0

### LINKED ISSUE #37: Sort duplicate references.

### LINKED ISSUE #38: Text without RefsDecl passes tests on refsDecl
See https://github.com/OpenGreekAndLatin/First1KGreek/blob/2068483b0d84076f7bef006be0263f4896a2a1d5/data/tlg4089/tlg004/tlg4089.tlg004.1st1K-grc1.xml

### LINKED ISSUE #39: Do shallow clone
Big repositories takes time to download. Reduce this like travis does.

### LINKED ISSUE #40: Check that URNs share the same spaces along work metadata file
Ie editions and translations have the same work urn

### PR DESCRIPTION
**Pull request by PonteIneptique**
- Migrated to MyCapytain >= 1.0.1 and its cleaner API ( Fixed #36 )
- Test on TEI Body instead of TEI Text for TEI Scheme according to Capitains.github.io/pages/guidelines.html ( Fixed #35 )
- Number of passages are now correctly computed ( Fixed #31 )
- Duplicate references are now sorted ( Fixed #37 )
- Remove Capitain checking for Text Units. Merged with Parsing ( Fixed #20 )
- Added new tests to fail refsDecl when there is no refsDecl ( Fixed #38 )
- Moved to shallow clone of depth 10 by default ( Fixed #39 )
- Checks that work has a language attribute ( Fixed #24 )
- Checks that the main urn (work or textgroup) relates to the path ( Fixed #23 )
- Added stronger support for URN tests for __cts__.xml ( Fixed #40 )
- Upgraded tests as needed
- Added coverage with subreference error testing

### CODE REVIEW

File: HookTest/units.py [hunk 1]
```diff
@@ -390,32 +455,39 @@ def unique_passage(self):
...
             yield False
 
-
     def has_urn(self):
         """ Test that a file has its urn saved
         """
         if self.xml is not None:
             if self.scheme == "tei":
-                urns = self.xml.xpath("//tei:text[starts-with(@n, 'urn:cts:')]", namespaces=TESTUnit.NS)
+                urns = self.xml.xpath("//tei:text/tei:body[starts-with(@n, 'urn:cts:')]", namespaces=TESTUnit.NS)
```
Comment (perseids-proxy-user) [inline HookTest/units.py]: Why this change here? Did we change the guidelines for this? Doesn't the URN belong on the text element?

File: HookTest/units.py [hunk 2]
```diff
@@ -390,32 +455,39 @@ def unique_passage(self):
         except Exception:
             yield False
 
-
     def has_urn(self):
         """ Test that a file has its urn saved
         """
         if self.xml is not None:
             if self.scheme == "tei":
-                urns = self.xml.xpath("//tei:text[starts-with(@n, 'urn:cts:')]", namespaces=TESTUnit.NS)
```
Comment (balmas) [inline HookTest/units.py]: why this change here? did we change the guidelines? doesn't the URN belong on the text element for TEI (i.e. not epidoc) files?

**Comment by balmas** [comment]
@PonteIneptique I just questioned the change in the positioning of the urn in non-epidoc TEI files, but otherwise this looks good to me.

File: HookTest/units.py (hunk 2 above)
Comment (balmas) [inline HookTest/units.py]: Ah, I see #35. Ok, not sure really which is right here, but if we have had that in the guidelines, I agree the test should at least match.
